lib/private/User/Session.php

```diff
@@ -780,7 +780,7 @@ private function validateToken($token, $user = null) {
 	 * Check if login names match
 	 */
 	private function validateTokenLoginName(?string $loginName, IToken $token): bool {
-		if ($token->getLoginName() !== $loginName) {
+		if (strtolower($token->getLoginName() ?? '') !== strtolower($loginName ?? '')) {
```

Comment:
`IToken::getLoginName` can never be null.

```suggestion
		if (strtolower($token->getLoginName()) !== strtolower($loginName ?? '')) {
```

Probably simpler with:
```suggestion
		if (strcasecmp($token->getLoginName(), $loginName ?? '') !== 0) {
```